sdk/lib/crt/stdio/file.c

```diff
@@ -2784,6 +2784,8 @@ int CDECL fclose(FILE* file)
 {
   int r, flag;
 
+  if(!file)
+    return EOF;
```

Comment:
According to [MSDN](https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/fclose-fcloseall?view=msvc-170#remarks):
>The fclose function closes stream. If stream is NULL, the invalid parameter handler is invoked, as described in [Parameter validation](https://learn.microsoft.com/en-us/cpp/c-runtime-library/parameter-validation?view=msvc-170). If execution is allowed to continue, fclose sets errno to EINVAL and returns EOF.

So what you are doing right now doesn't quite sound 100% right.